Sphinxsetup.sh

```diff
@@ -45,27 +55,26 @@ else
     SPHINX_VERSION="7.2.6"
 fi
 
-curl "$GET_PIP_URL" -o get-pip.py
-python3 get-pip.py
-rm -f get-pip.py
+PIP_USER_ARGUMENT=""
+if [ ${DISTRIBUTION_CODENAME} != 'noble' ]; then
+  curl "$GET_PIP_URL" -o get-pip.py
+  python3 get-pip.py
+  rm -f get-pip.py
+  PIP_USER_ARGUMENT="--user"
+fi
 
 # Install python packages using known working versions
 # Install sphinx with a specific docutils version
-python3 -m pip install --user --upgrade sphinx==${SPHINX_VERSION} "docutils<0.19"  "requests>=2.31.0" sphinx-tabs==3.*
-
+# Docutils version is for correct bullet point rendering. Can be rolled forward after theme is updated to >=0.5.1
+# See https://stackoverflow.com/a/68685753/2578171
 # lxml for parameter parsing:
-python3 -m pip install --user --upgrade lxml
-
 # Install sphinx theme from ArduPilot repository
-python3 -m pip install --user --upgrade git+https://github.com/ArduPilot/sphinx_rtd_theme.git
-
 # and youtube and video plugins:
 # This command might require a --force option if you have and older extension installed
 # Rerun Sphinxsetup.sh after doing that
-python3 -m pip install --user --upgrade git+https://github.com/ArduPilot/sphinxcontrib-youtube.git
-
+# and a parser to use getting posts from Discourse (forum) and insert in FrontEnd
 # Install flake8
-python3 -m pip install --user --upgrade flake8==3.7.9
+python3 -m pip install $PIP_USER_ARGUMENT --upgrade sphinx==${SPHINX_VERSION} "docutils<0.19" "requests>=2.31.0"  sphinx-tabs==3.* lxml git+https://github.com/ArduPilot/sphinx_rtd_theme.git git+https://github.com/ArduPilot/sphinxcontrib-youtube.git beautifulsoup4 flake8==3.7.9
```

Comment:
Please leave these as separate lines.

The efficiency is in no way important for this script, and it's easier to tell what's gone wrong with the separate lines.

It's also easier to tell exactly what's going on with the separate lines

Reply:
I put them together as pip has a much improved dependency resolver that can handle version conflicts when given all of the information. I can separate them out. I'm going to update #4642 next to move the list of packages to something common between setup scripts.

Reply:
I'm not sure I follow. With the current --upgrade option doing this on separate lines could result in different behavior as follow on pip install commands do not take into account the previously commanded pinning. Only when given all packages at once can the dependency resolver correctly do both updating and pinning.

Reply:
I would say that isn't a bad idea to regroup things together